### PR TITLE
Dynamic user agent

### PR DESCRIPTION
**Pull request by narekhovhannisyan**
## Motivation

The Mailtrap Node.js SDK needed to automatically detect when it's running within a Mailtrap MCP (Model Context Protocol) environment and use an appropriate User-Agent string. This improves environment detection, integration tracking, and provides better analytics for MCP-specific usage patterns.

## Changes

- Added dynamic User-Agent detection that automatically switches between `mailtrap-nodejs` and `mailtrap-mcp` based on runtime context
- Implemented comprehensive MCP context detection using multiple methods: main module filename, working directory, call stack analysis, and MCP runtime detection
- Added `MCP_USER_AGENT` configuration option to support MCP-specific User-Agent string
- Created comprehensive unit test suite with 12 test cases covering all detection scenarios and edge cases
- Updated `MailtrapClient` to use dynamic User-Agent selection instead of static configuration

## How to test

- [ ] Run `npm test -- src/__tests__/lib/get-agent.test.ts` to verify all 12 unit tests pass
- [ ] Test in regular Node.js environment - should use `mailtrap-nodejs` User-Agent
- [ ] Test in MCP environment with working directory containing "mailtrap-mcp" - should use `mailtrap-mcp` User-Agent
- [ ] Test in MCP runtime context (Claude Desktop) - should detect and use MCP User-Agent
- [ ] Verify User-Agent header in HTTP requests matches expected value for each environment
- [ ] Test error handling when `process.cwd()` throws exceptions
- [ ] Verify node_modules exclusion logic works correctly

## Summary by CodeRabbit

* **New Features**
  * Client now exposes the Suppressions API.
  * Added an optional configuration to set a custom User-Agent header.

* **Improvements**
  * Email payloads are encoded before sending to improve delivery reliability.

### CODE REVIEW

File: src/lib/get-agent.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    stack.includes("mailtrap-mcp") &&
+    !stack.includes("node_modules/mailtrap")
+  );
+}
+
+/**
+ * Gets the appropriate User-Agent string based on the current context.
+ * @returns The User-Agent string for the current context
+ */
+function getDynamicUserAgent(): string {
```
Comment (vittorius) [inline src/lib/get-agent.ts:65]: If you ask me, bringing the MCP concern to the SDK project is a violation of the SRP 🙂 I would approach this differently and allow specifying the user agent directly via a configuration setting of the SDK client. And, override this setting in the MCP server NPM package.
It's simpler, and it's not a big deal to allow users of the SDK to override the user agent; it's fairly easy to do this anyway.
WDYT?

Comment (narekhovhannisyan) [reply src/lib/get-agent.ts:65]: if allowing users to override user agent is not a problem, then its simple solution. I thought that we don't want to enable users to change the user agent.

Comment (leonid-shevtsov) [reply src/lib/get-agent.ts:65]: I agree with @vittorius, it's cleaner to allow to override user agent, and it might be useful for other purposes (for the users), too.

**Review by leonid-shevtsov — APPROVED** [review]
